### ISSUE TITLE
The published web app can no longer be used in accounts where API Access Control block unknown apps

### ISSUE THREAD

**Issue by NoSubstitute** [ericyd/gdrive-copy#191]
### Any specific steps to reproduce the issue?
Try to add the app client_id in API Access Control here. Click the _Add app_ link and supply the client_id.
https://admin.google.com/ac/owl/list?tab=configuredApps

client_id: 983475130545-e81q4d30k615fsr942krcejaajriv23a@developer.gserviceaccount.com

Since the client_id can not be found, it can't be added to the allowed list.

### Oath of good faith

* [X] I, the author of this Issue, do solemnly swear that I've looked _real hard_ for duplicate issues and I came up empty!!

**Comment by ericyd** [comment]
This is an issue with your GSuite account settings, I don't have control over this configuration

**Comment by santi-ifontana** [comment]
@NoSubstitute try with this ID: 983475130545-e81q4d30k615fsr942krcejaajriv23a.apps.googleusercontent.com

It worked for me.

**Comment by NoSubstitute** [comment]
> It worked for me.
Hello @santi-ifontana - thanks for trying to help.
Adding that, I get one step closer, but I'm still blocked.

![image](https://github.com/user-attachments/assets/f477d22d-7ccb-4deb-8598-5578a2a698dd)

This is the URL.
https://accounts.google.com/signin/oauth/danger?authuser=0&part=SomeUniqueValueHere&flowName=GeneralOAuthFlow&hl=sv&as=SNumbersHere%3AOtherNumbersHere&client_id=983475130545-e81q4d30k615fsr942krcejaajriv23a.apps.googleusercontent.com&rapt=SomeOtherUniqueValueHere

I am trying to limit access to only necessary things. So I didn't choose _Trusted_, but instead _Specific Google data_.

![image](https://github.com/user-attachments/assets/d3a9de2b-4676-4fbf-95f2-e9e1dace5bb2)

**Comment by santi-ifontana** [comment]
@NoSubstitute At first I also tried "Specific Google data" to limit the scope, but that did not work for me either.
Only "Trusted" seems to work. Also remember that you are pre-whitelisting, not actually giving permissions by selecting Trusted, only when you follow the wizard with your user is that you are allowing permissions into your account.

Please try again and let me know :)

**Comment by NoSubstitute** [comment]
Yeah, setting it to _Trusted_ works, but that makes the _Specific data_ setting pointless. 😞 
Thank you.

**Comment by santi-ifontana** [comment]
Yeah, I agree. I don't know why it does not work when you set the specific access, maybe because it is in **BETA** as displayed there.
Also, this additional step of grating additional access was not necessary, something happened on Google's side enforcing greater control.

**Comment by NoSubstitute** [comment]
Oh, enforcing granted access is by default for Workspace for Education for users Under-18, as Google forced the setting to _Block API access for all unknown third-party services_, which is something **_all_** organisations should have set as default a long time ago, when the feature was introduced in _API Access Control_.

It was one of the best days of my working life when I could just tick a single box, and no external _Login with Google_-junk was accessible, unless pre-approved by me.

**Comment by santi-ifontana** [comment]
That explains why you had this issue since 2023 and the rest of us just recently https://github.com/ericyd/gdrive-copy/issues/193

I didn't know, thanks for that clarification.
